ui/src/views/setting/ConfigurationRow.vue

```diff
@@ -0,0 +1,165 @@
+          <template #name="{ record }">
+            <b> {{record.displaytext }} </b> {{ ' (' + record.name + ')' }} <br/> {{ record.description }}
+          </template>
+
+          <template #value="{ record }">
+            <ConfigurationValue :configrecord="record" :loading="rowLoading" :configDisabled="configDisabled" @change-config="onChangeConfig" />
+          </template>
+
+          <template #expandedRowRender="{}" v-if="childrenConfigData.length > 0">
+            <a-table
+              size="small"
+              :showHeader="false"
+              :columns="childrenColumns"
+              :dataSource="childrenConfigData"
+              :rowKey="record => record.name"
+              :pagination="false"
+              :rowClassName="getRowClassName"
+              style="overflow-y: auto; margin-left: 10px" >
+
+              <template #displaytext="{ record }">
+                <ConfigurationRow :config="this.childrenConfigData" :configrecord="record" :loading="rowLoading" :configDisabled="!childrenConfigEnabled" />
+              </template>
+            </a-table>
+          </template>
+    </a-table>
+  </a-spin>
+</template>
+<script>
+import { api } from '@/api'
+import ConfigurationValue from './ConfigurationValue'
+
+export default {
+  name: 'ConfigurationRow',
+  components: {
+    ConfigurationValue
+  },
+  props: {
+    config: {
+      type: Object,
+      required: true
+    },
+    configrecord: {
+      type: Object,
+      required: true
+    },
+    loading: {
+      type: Boolean,
+      default: false
+    },
+    configDisabled: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      rowLoading: this.loading,
+      innerColumns: [
+        {
+          title: 'name',
+          dataIndex: 'name',
+          slots: { customRender: 'name' }
+        },
+        {
+          title: 'value',
+          dataIndex: 'value',
+          slots: { customRender: 'value' },
+          width: '25%'
+        }
+      ],
+      childrenColumns: [
+        {
+          title: 'Display Text',
+          dataIndex: 'displaytext',
+          slots: { customRender: 'displaytext' }
+        }
+      ],
+      childrenConfigData: [],
+      childrenConfigEnabled: true,
+      configrecords: [
+        this.configrecord
+      ]
+    }
+  },
+  created () {
+    this.fetchChildrenConfigData()
+  },
+  watch: {
+  },
+  methods: {
+    fetchChildrenConfigData () {
+      if (!this.isBooleanValue()) {
+        this.rowLoading = false
+        return
+      }
+      this.rowLoading = true
+      this.childrenConfigEnabled = (this.configrecord.value === 'true')
+      const params = {
+        parent: this.configrecord.name,
+        listAll: true
+      }
+      api('listConfigurations', params).then(response => {
+        this.childrenConfigData = response.listconfigurationsresponse.configuration
+        if (!this.childrenConfigData || this.childrenConfigData.length === 0) {
```

Comment:
I don't know what this paragraph is for. I think you can replace it like this `this.childrenConfigData = response?.listconfigurationsresponse?.configuration || []`